task.js

```diff
@@ -0,0 +1,231 @@
+
+// 6.- [ ] Create an object that counts the number of movies for each genre.
+
+const genreCount = filmsInfo.reduce((acc, film) => {
+    film.genre.forEach((genre) => {
+        acc[genre] = acc[genre] ? acc[genre] + 1 : 1;
+    });
+    return acc;
+}, {});
+
+console.log(filmsInfo);
+console.log(genreCount);
+console.log(filmsInfo);
+
+console.log("--------------------7-------------------");
+
+// 7. - [ ] Check if all movies have a rating higher than 7.
+
+console.log(filmsInfo);
+console.log(filmsInfo.every((film) => film.rating > 7));
+console.log(filmsInfo);
+
+console.log("--------------------8-------------------");
+
+// 8. - [ ] Calculate the average rating of all movies.
+
+console.log(filmsInfo);
+console.log(
+    filmsInfo.reduce((acc, film) => acc + film.rating, 0) / filmsInfo.length
+);
+console.log(filmsInfo);
+
+console.log("--------------------9-------------------");
+
+// 9. - [ ] Create an array of objects with only the title and rating of each movie.
+
+console.log(filmsInfo);
+console.log(
+    filmsInfo.map((film) => ({
+        title: film.title,
+        rating: film.rating,
+    }))
+);
+console.log(filmsInfo);
+
+console.log("--------------------10------------------");
+
+// 10. - [ ] Check if there is at least one movie in the "Sci-Fi" genre.
+
+console.log(filmsInfo);
+console.log(filmsInfo.some((film) => film.genre.includes("Sci-Fi")));
+console.log(filmsInfo);
+
+console.log("--------------------11------------------");
+
+// 11. - [ ] Check if there is at least one movie directed by "Quentin Tarantino".
+
+console.log(filmsInfo);
+console.log(filmsInfo.some((film) => film.director === "Quentin Tarantino"));
+console.log(filmsInfo);
+
+console.log("--------------------12------------------");
+
+// 12. - [ ] Check if all movies were released after 1990.
+
+console.log(filmsInfo);
+console.log(filmsInfo.every((film) => film.year > 1990));
+console.log(filmsInfo);
+
+console.log("--------------------13------------------");
+
+// 13. - [ ] Find the titles of all movies directed by "Christopher Nolan" that have a rating higher than 8.5, sorted by rating in descending order.
+
+console.log(filmsInfo);
+console.log(
+    filmsInfo
+        .filter(
+            (film) => film.director === "Christopher Nolan" && film.rating > 8.5
+        )
+        .map((film) => film.title)
+        .sort((a, b) => (a.rating - b.rating ? -1 : 1))
+);
+console.log(filmsInfo);
+
+console.log("--------------------14------------------");
+
+// 14. - [ ] Calculate the average rating of movies released before the year 2000 that belong to the "Drama" genre.
+const filteredMovies = filmsInfo.filter(
+    (film) => film.year < 2000 && film.genre.includes("Drama")
+);
+
+console.log(filmsInfo);
+console.log(filteredMovies);
+console.log(
+    (
+        filteredMovies.reduce((acc, film) => acc + film.rating, 0) /
+        filteredMovies.length
+    ).toFixed(2)
+);
+console.log(filmsInfo);
```

Comment:
You can do this within one chain without needing to save a reference of the filtered array.